doc/nrf/glossary.rst

```diff
@@ -0,0 +1,426 @@
+.. _glossary:
+
+Glossary
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+Anycast addressing
+   An addressing type that routes datagrams to a single member of a group of potential receivers that are all identified by the same destination address.
+   This is a one-to-nearest association.
+
+Access Port Protection (APPROTECT)
+   A register used to prevent read and write access to all CPU registers and memory-mapped addresses.
+
+API Result
+   Indicates success or failure of an API procedure.
+   In case of failure, a comprehensive error code indicating the cause or reason for the failure is provided.
+
+Application
+   An implementation example that includes a variety of libraries to implement a specific use case.
+
+Application Programming Interface (API)
+   A language and message format used by an application program to communicate with an operating system, application, or other services.
+
+Board
+   In Zephyr and |NCS|, a data structure for describing hardware on the board and its configuration files.
+
+Carrier Wave (CW)
+   A single-frequency electromagnetic wave that can be modulated in amplitude, frequency, or phase to convey information.
+
+Cat-M1
+   LTE-M User Equipment (UE) category with a single RX antenna, specified in 3GPP Release 13.
+
+Cat-NB1
+   Narrowband Internet of Things (NB-IoT) User Equipment (UE) category with 200 kHz UE bandwidth and a single RX antenna, specified in 3GPP Release 13.
+
+Chain of Trust
+   A sequence of properties identifying the trustworthiness of each layer in the system, all the way back to a property in the system referred to as *root of trust* (RoT).
+   A secure system depends on building and maintaining a chain of trust through all the layers in the system.
+   Each step in this chain guarantees that the next step can be trusted to have certain properties because any unauthorized modification of a subsequent step will be detected and the process halted.
+
+Clone
+   A local copy of a remote Git repository obtained with ``git clone``.
+
+Commissioning
+   See Provisioning.
```

Comment:
Provisioning does not say anything about commissioning. The term needs to be defined.

Reply:
Actually, this term is not very well defined, even in the specs I consulted for coming up with the definition of provisioning. Often, the meaning of these 2 terms overlap (especially in the NCS docs), and I could not figure out a clear-cut definition of it. @jciupis @kyberdin Do you have any input here?

Reply:
I'm only familiar with the way these terms are used in the Thread specification. There, commissioning is a process of authenticating a new, unknown device and joining it to the network. Provisioning is essentially an exchange of application-specific or vendor-specific data and performing additional configuration based on that. The exact meaning of these two terms may vary depending on the protocol, but I guess the above definitions are a good starting point.

Reply:
Fixed, in the context of the thread specification.